scenarios/appsec/waf/test_blocking.py

```diff
@@ -0,0 +1,204 @@
+      background: radial-gradient(circle at 26% 19%, #fff, #f4f7f9);
+      display: -webkit-box;
+      display: -ms-flexbox;
+      display: flex;
+      -webkit-box-pack: center;
+      -ms-flex-pack: center;
+      justify-content: center;
+      -webkit-box-align: center;
+      -ms-flex-align: center;
+      align-items: center;
+      -ms-flex-line-pack: center;
+      align-content: center;
+      width: 100%;
+      min-height: 100vh;
+      line-height: 1;
+      flex-direction: column
+    }
+
+    p {
+      display: block
+    }
+
+
+    main {
+      text-align: center;
+      flex: 1;
+      display: -webkit-box;
+      display: -ms-flexbox;
+      display: flex;
+      -webkit-box-pack: center;
+      -ms-flex-pack: center;
+      justify-content: center;
+      -webkit-box-align: center;
+      -ms-flex-align: center;
+      align-items: center;
+      -ms-flex-line-pack: center;
+      align-content: center;
+      flex-direction: column
+    }
+
+    p {
+      font-size: 18px;
+      line-height: normal;
+      color: #646464;
+      font-family: sans-serif;
+      font-weight: 400
+    }
+
+    a {
+      color: #4842b7
+    }
+
+    footer {
+      width: 100%;
+      text-align: center
+    }
+
+    footer p {
+      font-size: 16px
+    }
+  </style>
+</head>
+
+<body>
+<main>
+  <p>Sorry, you cannot access this page. Please contact the customer service team.</p>
+</main>
+<footer>
+  <p>Security provided by <a
+    href="https://www.datadoghq.com/product/security-platform/application-security-monitoring/"
+    target="_blank">Datadog</a></p>
+</footer>
+</body>
+
+</html>
+"""
+
+
+@released(dotnet="?", golang="?", nodejs="?", php_appsec="?", python="?", ruby="?")
+@released(
+    java={
+        "spring-boot": "0.112.0",
+        "sprint-boot-jetty": "0.112.0",
+        "spring-boot-undertow": "0.112.0",
+        "spring-boot-openliberty": "0.115.0",
+        "*": "?",
+    }
+)
+@coverage.basic
+class Test_Blocking(BaseTestCase):
+    """Blocking response is obtained when triggering a blocking rule"""
+
+    @bug(context.library < "java@0.115.0" and context.weblog_variant == "spring-boot-undertow", reason="npe")
+    def test_no_accept(self):
+        """Blocking without an accept header"""
+        r = self.weblog_get("/waf/", headers={"User-Agent": "Arachni/v1"})
+        assert r.status_code == 403
+        assert re.match("^application/json", r.headers["content-type"]) is not None
+        assert (
+            r.text == '{"errors": [{"title": "You\'ve been blocked", "detail": "Sorry, you cannot access '
+            'this page. Please contact the customer service team. Security provided by Datadog."}]}\n'
+        )
+
+    def test_blocking_appsec_blocked_tag(self):
+        """Tag ddappsec.blocked is set when blocking"""
+        r = self.weblog_get("/waf/", headers={"User-Agent": "Arachni/v1", "Accept": "*/*"})
+        assert r.status_code == 403
```

Comment:
```suggestion
        interfaces.library.add_assertion(r.status_code == 403)
```

And same for all others synchronous assertions